pansat/products/stations/igra.py

```diff
@@ -0,0 +1,379 @@
+"""
+pansat.products.soundings.igra
+=======================================
+
+This module defines the IGRA product class, which represents the global data product of radiosoundings: IGRA.
+
+
+"""
+
+import re
+import os
+from datetime import datetime, timedelta, date
+from pathlib import Path
+import pansat.download.providers as providers
+from pansat.products.product import Product
+import pandas as pd
+from math import radians, cos, sin, asin, sqrt
+import zipfile
+
+
+class NoAvailableProviderError(Exception):
+    """
+    Exception indicating that no suitable provider could be found for
+    a product.
+    """
+
+
+class IGRASoundings(Product):
+    """
+    The IGRA reanalysis class defines a generic interface for IGRA products.
+
+    Attributes:
+        name(``str``): name of product
+        locs(pd.DataFrame): pandas dataframe with metadata on stations
+                            (contains location ID, coordinates, and time period)
+
+        station(``pd.DataFrame``): pandas dataframe with metadata for chosen station
+        variable(``str`` ): variable to extract, if no station is given
+    """
+
+    def __init__(self, location=None, variable=None):
+        """
+        Args:
+
+        location(``str`` or tuple): station name or tuple with closest coordinates as float or int (lat,lon)
+
+        variable(``str`` ): variable to extract, if given monthly data of all stations will be downloaded
+
+        -- available variables:--
+        ghgt = Geopotential height
+        temp = Temperature
+        uwnd = Zonal wind component
+        vapr = Vapor pressure
+        vwnd = Meridional wind component
+        -----------------------------------
+        """
+        self.name = "igra-soundings"
+        provider = self._get_provider()
+        provider = provider(self)
+
+        destination = self.default_destination
+        destination.mkdir(parents=True, exist_ok=True)
+        # download meta data of all locations
+        downloaded = provider.download(
+            start=0,
+            end=0,
+            destination=destination,
+            base_url="ftp.ncdc.noaa.gov",
+            product_path="/pub/data/igra/",
+            files=["igra2-station-list.txt"],
+        )
+
+        self.variable = variable
+        self.station = location
+
+        # pandas data frame with all locations and meta information
+        self.locs = self.get_metadata()
+
+        # define column names of pandas dataframe with station info
+        colnames = ["ID", "lat", "lon", "n", "name", "start", "end", "nr"]
```

Comment:
What's ``n`` and ``nr``?

Reply:
n= elevation [m], nr = number of soundings in data record
I forgot to update because it was first unsure, but I verified it and now the columns have new names that are more clear